### PR TITLE
Fix warnings in timeseries.periodograms

### PR DESCRIPTION
**Pull request by bsipocz**
This is part of efforts of #7928, ~as well as to fix numpy 1.17 compatibility as noted in #8935~

### CODE REVIEW

**Comment by bsipocz** [comment]
While the Jake's cython was easily readable, I'm not sure I grasp what's going on in the python/cython/c layers for BLS, so one deprecation warning remains in there.
Also, this raises the issue that not all cases are covered in tests as the only deprecation is coming from the doctesting of the narrative docs.

@mhvk - could you please advise/push a fix to this branch?

File: astropy/timeseries/periodograms/lombscargle/implementations/cython_impl.pyx [hunk 1]
```diff
@@ -58,6 +58,10 @@ def lombscargle_cython(t, y, dy, frequency, normalization='standard',
         dy = 1
 
     t, y, dy = np.broadcast_arrays(t, y, dy)
+
+    t.flags.writeable = True
```
Comment (mhvk) [inline astropy/timeseries/periodograms/lombscargle/implementations/cython_impl.pyx]: I'm not sure it is right to just make the arrays writable - if any values are broadcast, something like `y -= ...` could very easily do the wrong thing. Will check.

**Comment by mhvk** [comment]
Looking into this more, this really seems an upstream issue, in numpy and/or cython; see https://github.com/numpy/numpy/issues/13929

**Comment by bsipocz** [comment]
Thanks @mhvk for hunting down the upstream issue. Once #8965 is merged I go ahead and remove the superfluous commit from here and rebase.

**Comment by pllim** [comment]
I am afraid straight backport to 2.x might not be possible due to the refactoring though. Are you sure you want to milestone this to 2.x? Aside from the milestone, I think this is good to merge.

**Comment by bsipocz** [comment]
Things just got moved around so git  should recognize it, but I will remilestone if it turns out to be problematic.

**Comment by bsipocz** [comment]
Affected file has been added in 3.0, so this cannot be backported to LTS.

**Pull request merged**